doc/bu-xthin-protocol.md

```diff
@@ -0,0 +1,382 @@
+thinblock strategy is designed to speed up the relay of blocks by using the transactions that already exist in the requester's 
+memory pool as a way to rebuild the block, rather than download it in its entirety. 
+
+Differing from other thinblock strategies, �Xtreme Thinblocks� uses simple bloom filtering and a new class of transactions to 
+ensure that almost all block requests only require a single round trip, which is essential for good performance on a system 
+where transaction rates are steadily climbing and only a synchronous network layer exists. In addition, Xtreme Thinblocks 
+uses a 64 bit transaction hash which further reduces thinblock size; in doing so a typical 1MB block can be reduced to a 10KB 
+to 25KB thinblock.
+
+
+2. High Level Design
+--------------------
+--------------------
+
+For various reasons memory pools are not in perfect sync which makes it difficult to reconstruct blocks in a fast and reliable 
+manner and with a single round trip. Creating a bloom filter at the time of sending the getdata request is an easy and fast way 
+around that problem which allows the remote node to know with high certainty what transactions are missing in the requester�s 
+memory pool and which need to be sent back in order to re-assemble the block.
+
+The reference implementation works in the following way:
+
+Node A is behind by one block and makes a thinblock request to Node B in the following way:
+
+1. Node A creates a bloom filter seeded with the contents of its memory pool.
+2. Node A sends the bloom filter along with a getdata request to Node B.
+3. Node B sends back a "thinblock" transaction which contains the block header information, all the transaction hashes that were 
+contained in the block, and any transactions that do not match the bloom filter which Node A had sent.
+4. Node A receives the "thinblock" and reconstructs the block using transactions that exist from its own memory pool as well as 
+the transactions that were supplied in the thinblock.?
+
+In the unlikely event that there are transactions still missing, they will be re-requested as follows.
+
+5. If there are still any transactions missing then a GET_XBLOCKTX message is sent from Node A.
+6. Node B upon receiving the GET_XBLOCKTX request, will get the missing transactions 
+from the block on disk rather than memory (in this way we can be sure the transactions can be accessed as they may already have been 
+purged from memory or they may have been unannounced). An XBLOCKTX message which contains the transactions is then sent back.
+
+Generally only one round trip is required to complete the block, but on occasion a transaction has to be re-requested initiating a 
+second round trip.
+
+In addition to the above, the following functionalities and configurations will be needed.
+
+- An XTHIN service bit
+
+- If the thinblocks feature is turned off then thinblocks will not be downloaded but requests for thinblocks will still be serviced.
```

Comment:
the feature is the service bit...(i'll make that more clear) but even that doesn't maybe make sense...i think, as far as the functionality, i was thinking in terms of being backwardly compatible in the future and reclaiming the service bit but maybe that's not realistic...maybe we should really have either end be able to turn it off completely, meaning if you are not running with bit ON then some other peer can't still request and xthin from you.  I supposes it's each implementation can do their own thing, it doesn't affect anyone else's...but it does make it more clear perhaps...thoughts?